django-stubs/template/base.pyi

```diff
@@ -73,7 +73,7 @@ class Template:
         engine: Optional[Engine] = ...,
     ) -> None: ...
     def __iter__(self) -> Iterator[Node]: ...
-    def render(self, context: Optional[Union[Context, Dict[str, Any]]]) -> SafeString: ...
+    def render(self, context: Context) -> SafeString: ...
```

Comment:
Note that this has to be compatible with `django.template.backends.base._EngineTemplate` protocol defined in django-stubs.

Reply:
Oh, wait. There are two `Template` classes in Django

`django.template.base.Template.render()` doesn't support dict, but `django.template.backends.django.Template.render()` is a wrapper class that does support `dict` and `None`.

`django.template.backends.django.DjangoTemplates.get_template()` wraps the first Template in the latter wrapper class. Confusing!